backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -445,19 +446,31 @@ export class ExperimentService {
         } catch (error) {
           throw new Error(`Error in updating experiment document "updateExperimentInDB" ${error}`);
         }
-
+        let defaultconditionCodeError = false;
```

Comment:
This variable is not required. Throw the error inside if condition

Reply:
Okay will update code.

Reply:
Wouldn't it be better to make a separate private function which is responsible for handling this part only?

```private hasConditionCodeDefault(conditions: Conditions[]): boolean {}```

If the function returns true, throw the error, otherwise proceed with further logic.

cc: @VivekFitkariwala

Reply:
Wouldn't it be better to make a separate private function which is responsible for handling this part only?

private hasConditionCodeDefault(conditions: Conditions[]): boolean {}

If the function returns true, throw the error, otherwise proceed with further logic.

cc: @VivekFitkariwala

Reply:
I agree. This is better way to do.

Reply:
Done accordingly